src/settings/AssistantSettings.tsx

```diff
@@ -0,0 +1,262 @@
+          id: uuidv4(),
+          name: '',
+          description: '',
+          imageUrl: '',
+        },
+      ],
+    }));
+  };
+
+  // Function to remove an assistant from the list
+  const handleRemoveAssistant = (index: number): void => {
+    onChange((prev) => ({
+      // Removing the assistant at the specified index
+      assistantsList: prev.assistantsList.filter((_, i) => i !== index),
+    }));
+  };
+
+  // Function to move an assistant up in the list
+  const handleMoveUp = (index: number): void => {
+    onChange((prev) => {
+      const updatedAssistants = [...prev.assistantsList];
+      const [movedAssistant] = updatedAssistants.splice(index, 1);
+      updatedAssistants.splice(index - 1, 0, movedAssistant);
+      return { assistantsList: updatedAssistants };
+    });
+  };
+
+  // Function to move an assistant down in the list
+  const handleMoveDown = (index: number): void => {
+    onChange((prev) => {
+      const updatedAssistants = [...prev.assistantsList];
+      const [movedAssistant] = updatedAssistants.splice(index, 1);
+      updatedAssistants.splice(index + 1, 0, movedAssistant);
+      return { assistantsList: updatedAssistants };
+    });
+  };
+
+  // Function to handle changes in the assistant's fields (name, description, image URL)
+  const handleChange = (
+    index: number,
+    field: keyof AssistantsSettingsType['assistantsList'][number],
+    value: string | number | boolean | (Agent & { type: AgentType.Assistant }),
+  ): void => {
+    const updatedAssistants = assistants.assistantsList.map((assistant, i) =>
+      // Updating the specified field of the assistant at the specified index
+      i === index ? { ...assistant, [field]: value } : assistant,
+    );
+
+    // Updating the state with the new list
+    onChange({ assistantsList: updatedAssistants });
+  };
+
+  return (
+    <Stack spacing={2}>
+      <Typography variant="h5">{t('SETTINGS.ASSISTANTS.TITLE')}</Typography>{' '}
+      <Stack
+        spacing={1}
+        py={2}
+        px={20}
+        border="1px solid #ccc"
+        borderRadius="8px"
+      >
+        {assistants.assistantsList.length === 0 ? (
+          <Alert
+            severity="warning"
+            sx={{
+              display: 'flex',
+              justifyContent: 'center',
+            }}
+          >
+            {t('SETTINGS.ASSISTANTS.CREATE')}{' '}
+          </Alert>
+        ) : (
+          // Mapping over the list of assistants and rendering a settings panel for each
+          assistants.assistantsList.map((assistant, index) => (
+            <Stack
+              key={index}
+              justifyContent="space-around"
+              direction="row"
+              spacing={2}
+              alignItems="center"
+              divider={
+                <Divider
+                  orientation="vertical"
+                  flexItem
+                  color={`#0${assistant.id.slice(0, 5)}`} // Unique color for each assistant's divider
```

Comment:
Since you use this twice, can you factor out?

Reply:
created constant
panelHex0To5